src/wasm/kotlin/java/arcs/Addressable.kt

```diff
@@ -0,0 +1,43 @@
+package arcs
+
+import arcs.AddressableMap.address2Addressable
+import arcs.AddressableMap.addressable2Address
+import arcs.AddressableMap.nextAddress
+
+/**
+ * Any object implementing this interface can be converted into a stable identifier. This
+ * could a platform specific pointer, identity hashcode, or id as long as it is guaranteed unique.
+ */
+interface Addressable
+
+typealias Address = Int
+
+internal object AddressableMap {
```

Comment:
This is a good suggestion. This will take more time -- while we are including kotlinx.atomicfu, it will take me a while to work out the kinks since we need to use this in a multi-platform way.

Reply:
I'll break out your suggestion into an issue, and we can implement this later.

Reply:
AtomicFu itself is probably not what you'd want here anyway (except for maybe `nextAddress`), I think using a mutex would be more appropriate. 

Though mutexes are probably also an issue with wasm?

Reply:
Yeah, I think they would be an issue for Wasm and JS. 

The approach would be to use the expected / actual mechanism to specify the platform appropriate implementation. 

For the Jvm, would using `synchronized` by sufficient on the toAddress toObject methods? 
Otherwise, I think I have a good understanding of what a mutex solution would look like.